storagemarket/contract_deal_monitor.go

```diff
@@ -90,106 +90,114 @@ func (c *ContractDealMonitor) Start(ctx context.Context) error {
 				}
 				return
 			case resp := <-responseCh:
-				err := func() error {
-					event := resp.Result.(map[string]interface{})
-					topicContractAddress := event["address"].(string)
-					topicDealProposalID := event["topics"].([]interface{})[1].(string)
-
-					// allowlist check
-					if len(c.cfg.AllowlistContracts) != 0 && !slices.Contains(c.cfg.AllowlistContracts, topicContractAddress) {
-						return fmt.Errorf("allowlist does not contain this contract address: %s", topicContractAddress)
-					}
-
-					res, err := c.getDealProposal(ctx, topicContractAddress, topicDealProposalID, fromEthAddr)
-					if err != nil {
-						return fmt.Errorf("eth call for get deal proposal failed: %w", err)
-					}
-
-					resParams, err := c.getExtraData(ctx, topicContractAddress, topicDealProposalID, fromEthAddr)
-					if err != nil {
-						return fmt.Errorf("eth call for extra data failed: %w", err)
-					}
-
-					var dpc market.DealProposal
-					err = dpc.UnmarshalCBOR(bytes.NewReader(res))
-					if err != nil {
-						return fmt.Errorf("cbor unmarshal failed: %w", err)
-					}
-
-					var pv1 types.ContractParamsVersion1
-					err = pv1.UnmarshalCBOR(bytes.NewReader(resParams))
-					if err != nil {
-						return fmt.Errorf("params cbor unmarshal failed: %w", err)
-					}
-
-					rootCidStr, err := dpc.Label.ToString()
-					if err != nil {
-						return fmt.Errorf("getting cid from label failed: %w", err)
-					}
-
-					rootCid, err := cid.Parse(rootCidStr)
-					if err != nil {
-						return fmt.Errorf("parsing cid failed: %w", err)
-					}
-
-					prop := market.DealProposal{
-						PieceCID:     dpc.PieceCID,
-						PieceSize:    dpc.PieceSize,
-						VerifiedDeal: dpc.VerifiedDeal,
-						Client:       dpc.Client,
-						Provider:     c.maddr,
-
-						Label: dpc.Label,
+				func() {
+					defer func() {
+						if r := recover(); r != nil {
```

Comment:
`recover` should catch any panics that come up due to misaligned `eth` namespace APIs with Lotus (see lines 101:103)